documentation/guide/writing-persistent-objects.rst

```diff
@@ -0,0 +1,568 @@
+==========================
+Writing persistent objects
+==========================
+
+In the :ref:`Tutorial <tutorial-label>`, we discussed the basics of
+implementing persistent objects by subclassing
+``persistent.Persistent``.  This is probably enough for 80% of
+persistent-object classes you write, but there are some other aspects
+of writing persistent classes you should be aware of.
+
+Access and modification
+=======================
+
+Two of the main jobs of the ``Persistent`` base class is to detect
+when an object has been accessed and when it has been modified.  When
+an object is accessed, its state may need to be loaded from the
+database.  When an object is modified, the modification needs to be
+saved if a transaction is committed.
+
+``Persistent`` detects object accesses by hooking into object
+attribute access and update.  In the case of object update, there
+may be other ways of modifying state that we need to make provision for.
+
+Rules of persistence
+====================
+
+When implementing persistent objects, be aware that an object's
+attributes should be :
+
+- immutable (such as strings or integers),
+
+- persistent (subclass Persistent), or
+
+- You need to take special precautions.
+
+If you modify a non-persistent mutable value of a persistent-object
+attribute, you need to mark the persistent object as changed yourself
+by setting ``_p_changed`` to True::
```

Comment:
In fact it is not set, but setting it to something triggers some functionality behind. If one reads ``_p_changed`` it does not return the value prior set. When I used this feature first several years ago I was kind of confused and only digging through the source helped me to understand what it is about. So I would phrase it:

```
If you modify a non-persistent mutable value of a persistent-object attribute, 
you need to tell the persistent object that it has changed by assigning ``_p_changed`` a value.
The assignment marks the object as modified 
(reading ``_p_changed`` does not tell you anything, since its just a trigger)::
```